easybuild/tools/easystack.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2020-2020 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Support for easybuild-ing from multiple easyconfigs based on
+information obtained from provided file (easystack) with build specifications.
+
+:author: Denis Kristak (Inuits)
+:author: Pavel Grochal (Inuits)
+"""
+
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.base import fancylogger
+try:
+    import yaml
+except ImportError:
+    pass
+_log = fancylogger.getLogger('easystack', fname=False)
+
+
+class Easystack(object):
+    """One class instance per easystack. General options + list of all SoftwareSpecs instances"""
+    def __init__(self):
+        self.easybuild_version = None
+        self.robot = False
+        self.software_list = []
+
+    def compose_ec_names(self):
+        """Returns a list of all easyconfig names"""
+        ec_names = []
+        for sw in self.software_list:
+            ec_to_append = '%s-%s-%s-%s.eb' % (str(sw.software), str(sw.version),
```

Comment:
Please use `det_full_ec_version`, rename `sw.software` to `sw.name`, and take into account `versionsufifx`.

Also the `str(...)` should not be needed here anymore (you should deal with that in `YamlEasystackParser.parse`):

```python
full_ec_version = det_full_ec_version({
    'toolchain': {'name': sw.toolchain_name, 'version': sw.toolchain_version},
    'version': sw.version,
    'versionsuffix': sw.versionsuffix,
})
ec_name = '%s-%s.eb' % (sw.name, full_ec_version)

Reply:
done